needle.pm

```diff
@@ -51,20 +52,29 @@ sub new {
     }
 
     my $self = {};
-    if (index($jsonfile, $bmwqemu::vars{PRJDIR}) == 0) {
-        $self->{file} = substr($jsonfile, length($bmwqemu::vars{PRJDIR}) + 1);
-    }
-    elsif (-f File::Spec->catfile($bmwqemu::vars{PRJDIR}, $jsonfile)) {
-        # json file path already relative
-        $self->{file} = $jsonfile;
-        $jsonfile = File::Spec->catfile($bmwqemu::vars{PRJDIR}, $jsonfile);
-    }
-    else {
-        die "Needle $jsonfile is not under project directory $bmwqemu::vars{PRJDIR}";
-    }
 
-    # $json->{file} contains path relative to $bmwqemu::vars{PRJDIR}
-    # $jsonfile contains absolute path within $bmwqemu::vars{PRJDIR}
+    # locate the needle file in one of the lookup directories ($bmwqemu::vars{PRJDIR} or the pool directory)
+    # - This code initializes $json->{file} so it contains the path relative to the lookup directory and re-assigns
+    #   $jsonfile to contain the absolute path.
+    # - The pool directory is considered to support loading custom needles with openqa-clone-custom-git-refspec.
+    my $pool_directory_checkout = (cwd . '/os-autoinst-needles-opensuse');    # global assumption: cwd == pooldir
```

Comment:
I thought you gave an example when you mentioned `os-autoinst-needles-opensuse` - I don't think this repo name should be in our code.

Reply:
Makes sense. Then we need a variable to pass additional search locations.

Reply:
So we? Can't this be a fixed name that the worker clones into?